### PR TITLE
Java- added binary literals and uppercase characters

### PR DESCRIPTION
**Pull request by vandiedakaf**
Added binary literals
Added uppercase literal characters

### CODE REVIEW

**Comment by jneen** [comment]
If you rebase this against master that test should pass.

**Comment by vandiedakaf** [comment]
Hi @jneen

I'm not sure if I've rebased and pushed the commits correctly (this all seems like black magic to me), yet it looks like this pull request is correct. What should I do if I wanted to make change to the code and not have that be part of this pull request?

File: lib/rouge/lexers/java.rb [hunk 1]
```diff
@@ -62,12 +62,14 @@ class Java < RegexLexer
         rule /[~^*!%&\[\](){}<>\|+=:;,.\/?-]/, Operator
 
         digit = /[0-9]_+[0-9]|[0-9]/
+        bin_digit = /[01]_+[01]|[01]/
         oct_digit = /[0-7]_+[0-7]|[0-7]/
-        hex_digit = /[0-9a-f]_+[0-9a-f]|[0-9a-f]/
+        hex_digit = /[0-9a-f]_+[0-9a-f]|[0-9a-f]/i
```
Comment (vandiedakaf) [inline lib/rouge/lexers/java.rb]: I just added the 'i' at the end.

File: lib/rouge/lexers/java.rb [hunk 2]
```diff
@@ -62,12 +62,14 @@ class Java < RegexLexer
...
         rule /#{digit}+\.#{digit}+([eE]#{digit}+)?[fd]?/, Num::Float
-        rule /0x#{hex_digit}+/, Num::Hex
+        rule /0b#{bin_digit}+/i, Num::Bin
```
Comment (vandiedakaf) [inline lib/rouge/lexers/java.rb]: Again just added the 'i' at the end.

**Comment by jneen** [comment]
Thanks!

**Pull request merged**